tests/integration/test_authd/test_authd_force_options.py

```diff
@@ -0,0 +1,252 @@
+                1) if insert_prev_agent_custom is present: previous input message is overwrite by the custom message
+                    (insert_prev_agent_custom: "OSSEC A:'user0' IP:'10.10.10.10'")
+                2) if insert_prev_agent_custom is not present: send the masage equals to input
+            - insert_random_pass_in_query:
+              "yes" if is needed add random pass to input query (for register with random pass cases)
+            - insert_hostname_in_query:
+              "yes" if is present add host name to input message
+    """
+    current_test = get_current_test()
+    test_case = force_options_tests[current_test]['test_case']
+
+    override_wazuh_conf(get_configuration)
+    for config in test_case:
+
+        # insert previous agent to force repeated case
+        try:
+            if config['insert_prev_agent'] == "yes":
+                try:
+                    response = send_message(config['insert_prev_agent_custom'])
+                except KeyError:
+                    response = send_message(config['input'])
+
+                # Prev output is expected
+                expected = "OSSEC K:'"
+                assert response, \
+                    'Failed connection previous insert for {}: {}'.format \
+                        (force_options_tests[current_test]['name'], config['input'])
+                assert response[:len(expected)] == expected, \
+                    "Failed response previous '{}': Input: {}".format \
+                        (force_options_tests[current_test]['name'], config['input'])
+                if expected == "OSSEC K:'":
+                    time.sleep(0.5)
+                    assert check_client_keys_file(response) == True, \
+                        "Failed test case '{}' checking previous client.keys : Input: {}".format \
+                            (force_options_tests[current_test]['name'], config['input'])
+        except KeyError:
+            pass
+
+        try:
+            if config['insert_hostname_in_query'] == 'yes':
+                config['input'] = config['input'].format(read_hostname())
+                config['output'] = config['output'].format(read_hostname())
+        except KeyError:
+            pass
+        except IndexError:
+            raise
+
+        # Output is expected
+        expected = config['output']
+        response = send_message(config['input'])
+        assert response, "Failed connection stage '{}'': '{}'".format \
+            (force_options_tests[current_test]['name'], config['input'])
+        if response[:len(expected)] != expected:
+            if config.get('expected_fail') == 'yes':
+                pytest.xfail("Test expected to fail by configuration")
+            else:
+                raise AssertionError("Failed test case '{}': Input: {}".format \
+                                         (force_options_tests[current_test]['name'], config['input']))
```

Comment:
`xfails` are not allowed. The only exception is a known bug with an associate issue in the repository. In this case, it looks that by the configuration itself wazuh should fail. If that is the case, please, ensure the error is the expected, removing xfail.

Reply:
The expected fails are related to this issue  https://github.com/wazuh/wazuh/issues/4965 .
A comment was added.